frontend/public/components/catalog-items.jsx

```diff
@@ -169,26 +209,104 @@ export class CatalogTileViewPage extends React.Component {
     return _.find(allCategories, { id: categoryID }).label;
   }
 
+  filterItems() {
+    const { filters } = this.state;
+    const { byName, byType } = filters;
+    const { items } = this.props;
+
+    let filteredItems = [];
+
+    if (byType.clusterServiceClass.active) {
+      filteredItems = _.filter(items, (item) => item.kind === byType.clusterServiceClass.value);
+    }
+
+    if (byType.imageStream.active) {
+      filteredItems = filteredItems.concat(_.filter(items, (item) => item.kind === byType.imageStream.value));
+    }
+
+    filteredItems = !_.isEmpty(filteredItems) ? filteredItems : items;
+
+    if (byName.active) {
+      const filterString = byName.value.toLowerCase();
+      return _.filter(filteredItems, item => item.tileName.toLowerCase().includes(filterString));
```

Comment:
We should be using `fuzzy` for filtering names

Reply:
Do we want to filter on description, too?

(I wouldn't use `fuzzy` for description because it will have a lot of unexpected matches.)

Reply:
> We should be using `fuzzy` for filtering names

I had that initially, but when I searched for 'net' it returned every item with 'n' 'e' or 't' in it's name.